src/counts.jl

```diff
@@ -450,5 +450,5 @@ Return a dictionary mapping each unique value in `x` to its proportion in `x`.
 If a vector of weights `wv` is provided, the proportion of weights is computed rather
 than the proportion of raw counts.
 """
-proportionmap(x::AbstractArray) = _normalize_countmap(countmap(x), length(x))
+proportionmap(x) = _normalize_countmap(countmap(x), length(collect(x)))
```

Comment:
Could we just count the total number of elements when building the `countmap`? It seems inefficient to materialize `x` only to obtain its length if we already iterate through it anyway.

Reply:
Something around sum(values(count(x))?

Reply:
No, I thought counting directly inside of `countmap`. But probably `sum(values, countmap(x))` would still be more efficient than using `collect(x)` if `x` is an iterator with a large number of elements.

Reply:
`` julia> @btime proportionmap(skipmissing(a))
  8.625 μs (27 allocations: 146.67 KiB)
Dict{Int64, Float64} with 4 entries:
  4 => 0.25
  2 => 0.25
  3 => 0.25
  1 => 0.25 ``
  
  `` julia> @btime proportionmap(skipmissing(a))
  316.667 ns (9 allocations: 1.08 KiB)
Dict{Int64, Float64} with 4 entries:
  4 => 0.25
  2 => 0.25
  3 => 0.25
  1 => 0.25 ``
  
  Looks like a significant improvement 🧐